internal/middleware/instrument.go

```diff
@@ -18,19 +18,189 @@
 package middleware
 
 import (
+	"encoding/json"
+	"fmt"
+	"github.com/caarlos0/env"
 	"github.com/gorilla/mux"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promauto"
 	"net/http"
 	"strconv"
+	"strings"
 	"time"
 )
 
+// urlpattern
+// [{"url":["url1,url2"], "label":{k1:v1,k2:v2}},
+// {"url":["url4,url5"], "label":{k5:v6,k1:v8}}
+// ]
+
+type HttpLabels struct {
+	UrlPaths string `env:"URL_PATHS" envDefault:"[]"`
+}
+
+var UrlLabelsMapping map[string]map[string]string
+var UniqueKeys []string
+
+const (
+	path   string = "path"
+	method string = "method"
+	status string = "status"
+)
+
+func GetHttpLabels() (*HttpLabels, error) {
+	cfg := &HttpLabels{}
+	err := env.Parse(cfg)
+	return cfg, err
+}
+
+/*
+getLabels processes URL and label information, storing mappings and unique keys.
+
+Example:
+Suppose the `GetHttpLabels` function returns the following data:
+
+	{
+	    "UrlPaths": `[
+	        {
+	            "url": ["http://example.com", "http://example.org"],
+	            "label": {"name": "Example", "type": "Website", "key1": "value1"}
+	        },
+	        {
+	            "url": ["http://example.com, http://example.net"],
+	            "label": {"name": "Example2", "type": "Site"}
+	        }
+	    ]`
+	}
+
+Output:
+Unique Keys:
+[
+
+	"path",
+	"method",
+	"status",
+	"name",
+	"type",
+	"key1"
+
+]
+URL Labels Mapping:
+
+	{
+	    "http://example.com": {"name": "Example2", "type": "Site", "key1": "value1"},
+	    "http://example.org": {"name": "Example", "type": "Website", "key1": "value1"},
+	    "http://example.net": {"name": "Example2", "type": "Site"}
+	}
+
+Explanation:
+1. The function calls `GetHttpLabels` to retrieve JSON data containing URL and label information.
+2. It unmarshals the JSON data into a slice of maps.
+3. It initializes maps to store URL to label mappings and unique label keys.
+4. It iterates over the data to process URLs and labels, merging them into `urlMappings`.
+5. It also collects unique keys while processing labels.
+6. The global variables `UrlLabelsMapping` and `UniqueKeys` are updated with the mappings and unique keys, respectively.
+*/
+func getLabels() []string {
```

Comment:
Since the URL vs Label vs Keys mapping that is coming from CM is in a particular format, This is the simplest way to extract required data from them.

` urlpattern
[{"url":["url1,url2"], "label":{k1:v1,k2:v2}},
 {"url":["url4,url81"], "label":{k5:v6,k1:v8}},
 {"url":["url9,url67"], "label":{k5:v6,k1:v8}}
 {"url":["url10,url35"], "label":{k5:v6,k1:v8}}
 {"url":["url15,url31,url1,url2"], "label":{k5:v6,k1:v8}}
 {"url":["url18,url27,url1,url2"], "label":{k5:v6,k1:v8}}
 ]`

Reply:
done